File: 6-11(June Challange 2021)/q2a.cpp
#include <iostream>

using namespace std;

int main(int argc, char const *argv[])
{
    int N;
    cin>>N;

    for (int j = 0; j < N; j++)
    {
        int D,d,P,Q;
        cin>>D>>d>>P>>Q;

        int rate = P;
        int i=0;
        int money = 0;

        while(i<D){
            money += rate;
            i+=1;
            if (i%d==0){
                rate+=Q;
            }
        }

        cout<<money<<endl;
    }
    


    return 0;
}

File: 6-11(June Challange 2021)/q2a.py
for _ in range(int(input())):
    D,d,P,Q = map(int,input().split(" "))

    rate = P
    i=0
    money = 0
    while(i<D):
        money += rate
        i+=1
        if (i%d==0):
            rate+=Q

    print(money)
        